flexget/components/sites/sites/filelist_api.py

```diff
@@ -0,0 +1,180 @@
+
+            raise plugin.PluginError(
+                f'FileList request failed; err code: {response.status_code}; err msg: `{error_message}`'
+            )
+
+        return response
+
+    def dump_entry(self, entry):
+        entry_items = {key: value for key, value in entry.items()}
+        logger.debug(f'entry_items={entry_items}')
+
+    @plugin.internet(logger)
+    def search(self, task, entry, config):
+        """
+            Search for entries on FileList
+        """
+        entries = list()
+
+        # TODO remove later
+        self.dump_entry(entry)
+
+        # mandatory params
+        params = {'username': config['username'], 'passkey': config['passkey'], 'action': 'search-torrents'}
+
+        # category
+        if config.get('category'):
+            params['category'] = (
+                ','.join([self.valid_categories[cat] for cat in config['category']])
+                if isinstance(config.get('category'), list)
+                else self.valid_categories[config.get('category')]
+            )
+
+        # extras: internal release, moderated torrent, freeleech
+        for extra_name, extra_value in self.valid_extras.items():
+            if config.get(extra_name):
+                params[extra_name] = extra_value
+
+        # set season/episode if series
+        if entry.get('series_episode'):
+            params['episode'] = entry.get('series_episode')
+        if entry.get('series_season'):
+            params['season'] = entry.get('series_season')
+
+        for search_title in entry.get('search_strings', [entry.get('title')]):
+
+            if entry.get('imdb_id'):
+                params['type'] = 'imdb'
+                params['query'] = entry.get('imdb_id')
+                params['name'] = search_title
+            else:
+                params['type'] = 'name'
+                params['query'] = search_title
+
+            # TODO remove later
+            logger.debug(f'get params: {params}')
+
+            # get them results
+            try:
+                response = self.get(self.api_url, params)
+                # TODO remove later
+                logger.debug(f'Requesting: {response.url}')
+            except RequestException as e:
+                raise plugin.PluginError(f'FileList request failed badly! {e}')
+
+            results = response.json()
+            if not results:
+                logger.info(f'No torrent found on Filelist for `{search_title}`')
+            else:
+                logger.info(f'I have found {len(results)} torrent(s) on Filelist for `{search_title}`')
+
+            for result in results:
+                entry = Entry()
+
+                entry['title'] = result['name']
+                entry['url'] = result['download_link']
+                entry['imdb'] = result['imdb']
+                # size is returned in bytes but expected in MiB
+                entry['content_size'] = result['size'] / 2 ** 20
```

Comment:
We have a util for this

Reply:
I have only found this https://github.com/Flexget/Flexget/blob/e33eed8f1a508555aeaffa3bf3b24600a8fea420/flexget/utils/tools.py#L388 however that takes a string as input

Reply:
I guess that fine